src/complex/Common/BoundingBox.hpp

```diff
@@ -81,145 +95,62 @@ class BoundingBox
+   * @brief Sets the Min point
    * @return
    */
-  std::array<ValueType, 3> sideLengths() const
+  void setMinPoint(PointType min)
   {
-    return {m_Bounds[3] - m_Bounds[0], m_Bounds[4] - m_Bounds[1], m_Bounds[5] - m_Bounds[2]};
+    m_Lower = min;
   }
 
   /**
-   * @brief compute the center of the bounding box.
+   * @brief Sets the Max point
    * @return
    */
-  std::array<ValueType, 3> center() const
-  {
-    std::array<ValueType, 3> boxSize = sideLengths();
-    return {m_Bounds[0] + boxSize[0] / 2.0, m_Bounds[1] + boxSize[1] / 2.0, m_Bounds[2] + boxSize[2] / 2.0};
-  }
-
-  /**
-   * @brief Returns the current minimum X value.
-   * @return ValueType
-   */
-  ValueType getMinX() const
-  {
-    return m_Bounds[0];
-  }
-
-  /**
-   * @brief Returns the current maximum X value.
-   * @return ValueType
-   */
-  ValueType getMaxX() const
-  {
-    return m_Bounds[3];
-  }
-
-  /**
-   * @brief Returns the current minimum Y value.
-   * @return ValueType
-   */
-  ValueType getMinY() const
-  {
-    return m_Bounds[1];
-  }
-
-  /**
-   * @brief Returns the current maximum Y value.
-   * @return ValueType
-   */
-  ValueType getMaxY() const
-  {
-    return m_Bounds[4];
-  }
-
-  /**
-   * @brief Returns the current minimum Z value.
-   * @return ValueType
-   */
-  ValueType getMinZ() const
-  {
-    return m_Bounds[2];
-  }
-
-  /**
-   * @brief Returns the current maximum Z value.
-   * @return ValueType
-   */
-  ValueType getMaxZ() const
+  void setMaxPoint(PointType max)
   {
-    return m_Bounds[5];
+    m_Upper = max;
   }
 
   /**
-   * @brief Sets the current minimum X value.
-   * @param value
-   */
-  void setMinX(ValueType value)
-  {
-    m_Bounds[0] = value;
-  }
-
-  /**
-   * @brief Sets the current maximum X value.
-   * @param value
-   */
-  void setMaxX(ValueType value)
-  {
-    m_Bounds[3] = value;
-  }
-
-  /**
-   * @brief Sets the current minimum Y value.
-   * @param value
-   */
-  void setMinY(ValueType value)
-  {
-    m_Bounds[1] = value;
-  }
-
-  /**
-   * @brief Sets the current maximum Y value.
-   * @param value
-   */
-  void setMaxY(ValueType value)
-  {
-    m_Bounds[4] = value;
-  }
-
-  /**
-   * @brief Sets the current minimum Z value.
-   * @param value
+   * @brief Compute the length of each side of the bounding box
+   * @return
    */
-  void setMinZ(ValueType value)
+  PointType sideLengths() const
   {
-    m_Bounds[2] = value;
+    return {m_Upper - m_Lower};
   }
 
   /**
-   * @brief Sets the current maximum Z value.
-   * @param value
+   * @brief compute the center of the bounding box.
+   * @return
    */
-  void setMaxZ(ValueType value)
+  PointType center() const
   {
-    m_Bounds[5] = value;
+    PointType boxSize = sideLengths();
+    if constexpr(std::is_same_v<Point<ValueType>, Point3D<ValueType>>)
+    {
+      return {m_Lower[0] + boxSize[0] / 2.0, m_Lower[1] + boxSize[1] / 2.0, m_Lower[2] + boxSize[2] / 2.0};
```

Comment:
Maybe we should have a division operator on `Vec`. I see we have multiplication already.